avldb/src/main/scala/scorex/core/serialization/ManifestSerializer.scala

```diff
@@ -0,0 +1,69 @@
+package scorex.core.serialization
+
+import com.google.common.primitives.Ints
+import scorex.crypto.authds.avltree.batch.Constants.DigestType
+import scorex.crypto.authds.avltree.batch.serialization.{BatchAVLProverManifest, ProxyInternalNode}
+import scorex.crypto.authds.avltree.batch.{InternalProverNode, ProverLeaf, ProverNodes, VersionedLDBAVLStorage}
+import scorex.util.serialization.{Reader, Writer}
+
+/**
+  * Serializer of manifest, a tree which is cut at some `manifestDepth` from root
+  */
+class ManifestSerializer(manifestDepth: Byte) extends ErgoSerializer[BatchAVLProverManifest[DigestType]] {
+  private val nodeSerializer = VersionedLDBAVLStorage.noStoreSerializer
+
+  override def serialize(manifest: BatchAVLProverManifest[DigestType], w: Writer): Unit = {
+    val height = manifest.rootHeight
+    w.putBytes(Ints.toByteArray(height))
```

Comment:
That would require VersionedLDBAVLStorage.dumpSnapshot to work with Writer somehow, but I do not see an efficient way to do that.

Reply:
ManifestSerializer is serializing an instance of `Manifest`, dumpSnapshot is going through the database node-by-node and so do not extract manifest and subtrees directly, to avoid allocations(the structure is about hundreds of megabytes and will be gigabytes in future)